### PR TITLE
Add OpenJFX 11

### PR DESCRIPTION
**Pull request by claudionieder**
After making all changes to the cask:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] The commit message includes the cask’s name and version.
- [X] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [X] Named the cask according to the [token reference].
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked there are no [open pull requests] for the same cask.
- [X] Checked the cask was not [already refused].
- [X] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/openjfx.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  name 'OpenJFX'
+  homepage 'https://openjfx.io/'
+
+  postflight do
+    system_command '/bin/cp',
+                   args: ['-rp', staged_path.to_s, '/usr/local/lib/OpenJFX']
+  end
+
+  uninstall_postflight do
+    next unless File.exist?('/usr/local/lib/OpenJFX')
```
Comment (commitay) [inline Casks/openjfx.rb]: Remove this, it isn't needed for this cask.

File: Casks/openjfx.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    system_command '/bin/rm',
+                   args: ['-rf', '/usr/local/lib/OpenJFX']
+  end
+end
```
Comment (commitay) [inline Casks/openjfx.rb]: Add a caveat:
https://github.com/Homebrew/homebrew-cask/blob/6ec7d441cd810a588a1ba24f4d77da636c987928/Casks/gpg-suite.rb#L73-L75

**Comment by claudionieder** [comment]
Hi,

as I am new to this, please allow me two questions:

1. I have added the caveat and now when it is installed it puts out a warning. Even though it is unsubstantiated, because brew doctor does not complain about anything this made me wonder, if my selection to install it to /usr/local/lib/OpenJFX is a wise one? Do you think there would be a better place where to put OpenJFX?

2. As I am not so versed in git and especially github: After I have fixed the issues is it the same

	git commit -v
	git push <my-github-username> my-new-cask
	Hit "Compare & pull request" in github

sequence as described in adding_a_cask.md or do I need to modify any of above commands?

Thank you very much for your help,
claudio

File: Casks/openjfx.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  version '11'
+  sha256 'adabd9332993519e8ad69c5412a9acb00443e3b80a15c3d372a99b064a94873c'
+
+  # download2.gluonhq.com/openjfx/ was verified as official when first introduced to the cask
+  url "http://download2.gluonhq.com/openjfx/#{version}/openjfx-#{version}_osx-x64_bin-sdk.zip"
+  name 'OpenJFX'
+  homepage 'https://openjfx.io/'
+
+  postflight do
+    system_command '/bin/cp',
```
Comment (vitorgalvao) [inline Casks/openjfx.rb]: Use ruby-native methods (`File` or `FileUtils`).

File: Casks/openjfx.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                   args: ['-rp', staged_path.to_s, '/usr/local/lib/OpenJFX']
+  end
+
+  uninstall_postflight do
+    next unless File.exist?('/usr/local/lib/OpenJFX')
+
+    system_command '/bin/rm',
```
Comment (vitorgalvao) [inline Casks/openjfx.rb]: Use ruby-native methods (`File` or `FileUtils`).

File: Casks/openjfx.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+cask 'openjfx' do
+  version '11'
+  sha256 'adabd9332993519e8ad69c5412a9acb00443e3b80a15c3d372a99b064a94873c'
+
+  # download2.gluonhq.com/openjfx/ was verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/openjfx.rb]: Remove the last slash.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
> wonder, if my selection to install it to /usr/local/lib/OpenJFX is a wise one?

What does the official documentation recommend?

**Comment by claudionieder** [comment]
The official documentation simply says "Download an appropriate JavaFX runtime for your operating system and unzip them to a desired location." It leaves the choice to the user.

File: Casks/openjfx.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  url "http://download2.gluonhq.com/openjfx/#{version}/openjfx-#{version}_osx-x64_bin-sdk.zip"
+  name 'OpenJFX'
+  homepage 'https://openjfx.io/'
+
+  postflight do
+    FileUtils.cp_r(staged_path.to_s, '/usr/local/lib/OpenJFX')
+  end
+
+  uninstall_postflight do
+    FileUtils.rm_rf ['/usr/local/lib/OpenJFX']
```
Comment (vitorgalvao) [inline Casks/openjfx.rb]: Use parenthesis (consistent with `postflight`).

**Review by vitorgalvao — APPROVED** [review]

**Comment by claudionieder** [comment]
In the discussion thread where I originally asked if anybody is already working on this (https://discourse.brew.sh/t/is-anybody-already-working-on-an-openjfx-cask/3083) it was now pointed out, that OpenJFX could also be made a maven/gradle/your-java-build-tool-of-choice dependency and thus a cask like this is not needed. I verified this, and therefore I agree with those opinions and think that this cask shall not be pursued further... even though I arrived a parenthesis near to completion :-)

Do I have to do something to declare this dead, or how is such a case handled?

**Comment by commitay** [comment]
Thanks for the PR nonetheless!

**Pull request closed without merging**